salt/tests/unit/modules/test_metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,221 @@
+    ):
+        """
+        Tests the return of `get_kubeconfig` function
+        """
+        pillar_dict = {}
+        if api_server_pillar:
+            pillar_dict = {'metalk8s': {'api_server': api_server_pillar}}
+
+        config_options_dict = {}
+        if config_options:
+            config_options_dict = config_options
+
+        config_options_mock = MagicMock(side_effect=config_options_dict.get)
+
+        salt_dict = {
+            'config.option': config_options_mock,
+        }
+
+        with patch.dict(metalk8s_kubernetes_utils.__pillar__, pillar_dict), \
+                patch.dict(metalk8s_kubernetes_utils.__salt__, salt_dict):
+            self.assertEqual(
+                tuple(result),
+                metalk8s_kubernetes_utils.get_kubeconfig(**kwargs)
+            )
+
+    @parameterized.expand([
+        (['my_kubeconfig.conf', 'my_context'], {"major": "1", "go_version": "go1.13.8"}),
+        (['my_kubeconfig.conf', 'my_context'], 'Failed to get version info', True, True),
+    ])
+    def test_get_version_info(
+        self,
+        kubeconfig,
+        result,
+        error=False,
+        raises=False,
+        **kwargs
+    ):
+        """
+        Tests the return of `get_version_info` function
+        """
+        kubeconfig_mock = MagicMock(return_value=tuple(kubeconfig))
+
+        api_instance_mock = MagicMock()
+        if error:
+            api_instance_mock.return_value.get_code.side_effect = \
+                ApiException('Failed to get version info')
+
+        api_instance_mock.return_value.get_code.return_value.to_dict \
+            .return_value = result
+
+        with patch(
+            "metalk8s_kubernetes_utils.get_kubeconfig", kubeconfig_mock
+            ), patch(
+                "kubernetes.config.new_client_from_config", MagicMock()
+                ), patch("kubernetes.client.VersionApi", api_instance_mock):
+            if raises:
+                self.assertRaisesRegexp(
+                    CommandExecutionError,
+                    result,
+                    metalk8s_kubernetes_utils.get_version_info
+                )
+            else:
+                self.assertEqual(
+                    metalk8s_kubernetes_utils.get_version_info(**kwargs),
+                    result
+                )
+            api_instance_mock.return_value.get_code.assert_called()
+
+    @parameterized.expand([
+        ({"go_version": "my_go_version"}, True),
+        (None, False, True, True),
+    ])
+    def test_ping(
+        self,
+        version_info,
+        result,
+        error=False,
+        raises=False,
+    ):
+        """
+        Tests the return of `ping` function
+        """
+        version_info_mock = MagicMock()
+        if error:
+            version_info_mock.side_effect = CommandExecutionError(
+                'Failed to get version info'
+            )
+        version_info_mock.return_value.get_version_info.return_value = \
+            version_info
```

Comment:
As i told you on slack it's wrong here you will return this value when you will call `get_version_info().get_version_info()`

And as I told you in other comment we do not care about the `get_version_info` output we just want this function to raise in some cases